File: app/models/gate.py
"""
InfecSure - ICNO Validation Gate models.
"""

from __future__ import annotations

from typing import Literal, Optional

from pydantic import BaseModel


class GateValidationRequest(BaseModel):
    alert_id: str
    decision: Literal["approve", "reject"]
    icno_notes: Optional[str] = None

File: InfecSure_Frontend/src/api/gate.ts
import { api } from "./client";

export async function listGatePending() {
  const { data } = await api.get("/gate/pending");
  return data;
}

export async function validateGateItem(alertId: string, decision: "approve" | "reject", notes?: string) {
  const { data } = await api.post("/gate/validate", {
    alert_id: alertId,
    decision,
    icno_notes: notes || ""
  });
  return data;
}
